app/src/main/java/com/cooper/wheellog/utils/InMotionAdapter.java

```diff
@@ -1237,35 +1222,51 @@ boolean parseSlowInfoMessage() {
 
         ByteArrayOutputStream buffer = new ByteArrayOutputStream();
         int oldc = 0;
-        int oldestc = 0;
+        int len_p = 0;
+        int len_ex = 0;
+
         UnpackerState state = UnpackerState.unknown;
 
         byte[] getBuffer() {
             return buffer.toByteArray();
         }
 
         boolean addChar(int c) {
-            if (state == UnpackerState.collecting) {
-                buffer.write(c);
-                if (c == (byte) 0x55 && oldc == (byte) 0x55 && oldestc != (byte) 0xA5) {
-                    state = UnpackerState.done;
-                    updateStep = 0;
-                    oldc = 0;
-                    Timber.i("Step reset");
-                    return true;
-                }
-            } else {
-                if (c == (byte) 0xAA && oldc == (byte) 0xAA && oldestc != (byte) 0xA5) {
-                    buffer = new ByteArrayOutputStream();
-                    buffer.write(0xAA);
-                    buffer.write(0xAA);
-                    state = UnpackerState.collecting;
+            if (c != (byte) 0xA5 || oldc == (byte) 0xA5) {
+
+
+                if (state == UnpackerState.collecting) {
+                    buffer.write(c);
+                    int sz = buffer.size();
+                    if (sz == 7) len_ex = c & 0xFF;
+                    if (sz == 15) len_p = c & 0xFF;
```

Comment:
одновременно эти два условия не выполняются. Пожтому лучше добавить
```suggestion
                    else if (sz == 15) len_p = c & 0xFF;
```